src/Components/Facility/InventoryLog.tsx

```diff
@@ -1,123 +1,90 @@
-      setIsLoading(true);
-      const res = await dispatchAction(
-        getInventoryLog(facilityId, { item, limit, offset })
-      );
-      if (!status.aborted) {
-        if (res?.data) {
-          setInventory(res.data.results);
-          setCurrentStock(res.data.results[0].current_stock);
-          setTotalCount(res.data.count);
-          setItemName(res.data.results[0].item_object.name);
-        }
-        setIsLoading(false);
-      }
-    },
-    [dispatchAction, offset, facilityId]
+  const { data: InventoryLog, refetch: InventoryFetch } = useQuery(
+    routes.getInventoryLog,
+    {
+      pathParams: {
+        facilityId: facilityId,
+      },
+      query: {
+        item,
+        limit,
+        offset,
+      },
+      prefetch: facilityId !== undefined || !!offset,
+    }
   );
 
-  useEffect(() => {
-    async function fetchFacilityName() {
-      if (facilityId) {
-        const res = await dispatchAction(getAnyFacility(facilityId));
-
-        setFacilityName(res?.data?.name || "");
-      } else {
-        setFacilityName("");
-      }
-    }
-    fetchFacilityName();
-  }, [dispatchAction, facilityId]);
+  const { data: facilityObject } = useQuery(routes.getAnyFacility, {
+    pathParams: { id: facilityId },
+    prefetch: !!facilityId,
+  });
 
   const flagFacility = async (id: string) => {
     setSaving(true);
-    const res = await dispatchAction(
-      flagInventoryItem({ facility_external_id: facilityId, external_id: id })
-    );
-
-    if (res && res.status === 204) {
-      Notification.Success({
-        msg: "Updated Successfully",
-      });
-      fetchData({ aborted: false });
-    }
+    await request(routes.flagInventoryItem, {
+      pathParams: { facility_external_id: facilityId, external_id: id },
+      onResponse: ({ res }) => {
+        if (res?.ok) {
+          Notification.Success({
+            msg: "Updated Successfully",
+          });
+        }
+        InventoryFetch();
+      },
+    });
     setSaving(false);
   };
 
   const removeLastInventoryLog = async (id: any) => {
     setSaving(true);
-    const res = await dispatchAction(
-      deleteLastInventoryLog({
-        facility_external_id: facilityId,
-        id: id,
-      })
-    );
-
-    if (res?.status === 201) {
-      Notification.Success({
-        msg: "Last entry deleted Successfully",
-      });
-      fetchData({ aborted: false });
-    } else {
-      Notification.Error({
-        msg: "Error while deleting last entry: " + (res?.data?.detail || ""),
-      });
-    }
+    await request(routes.deleteLastInventoryLog, {
+      pathParams: { facility_external_id: facilityId, id: id },
+      onResponse: ({ res, data }) => {
+        if (res?.ok) {
+          Notification.Success({
+            msg: "Last entry deleted Successfully",
+          });
+          InventoryFetch();
+        } else {
+          Notification.Error({
+            msg: "Error while deleting last entry: " + (data?.detail || ""),
+          });
```

Comment:
Are you sure non 2xx responses are present in `data`?
```suggestion
          Notification.Error({
            msg: "Error while deleting last entry: " + (error.detail || ""),
          });
```

Reply:
You maybe actually be able skip triggering a notification since useQuery/request will dispatch notifications for you for certain status codes.